src/Missings.jl

```diff
@@ -165,4 +166,46 @@ function levels(x)
     levs
 end
 
+struct Conditional{P,X,Y} <: Function end
+
+"""
+    conditional(predicate, x, y)
```

Comment:
What's the interest of having such a generic function here? Wouldn't it be simpler to hardcode the behavior of `passmissing`?

Reply:
The first reason is that I understand that this is "testing phase" of the concept. The function could be useful in Base for other reasons in the future, e.g. to provide similar functionality for `nothing`.

The second reason is that, as I have indicated in the main thread https://github.com/JuliaLang/julia/pull/26661, even for `missing` a user might want a specific behavior. For instance:

`passmissing(string)(10, pad=missing)`

would produce `missing`, but someone might want `missing` in `pad`  to be treated in a different way (e.g. throw an error) and write something like

```
condidional((x;kw...) -> ismissing(x), (x;kw...) -> missing, string)
```

But I will not stick with it if people do not find it useful.

Reply:
I see your point, but Missings.jl only deals with missing values, so `nothing` is out of scope here. And given that the amount of code is small, even duplication wouldn't be too bad (actually, the docstring for `conditional` takes most of the lines).

Reply:
OK - so I understand we implement only a specialized `passmissing` for now? If yes I will amend the PR.

Reply:
I'd say so, but you can wait for more input from others.